hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,295 @@
+          m_R(R), m_density(density), m_seed(seed), m_picks(m_exec_conf), m_bc(bc)
+          {}
+
+        //! Implementation of the streaming rule
+        virtual void stream(unsigned int timestep);
+
+    protected:
+        std::shared_ptr<::Variant> m_R;      //!< Radius of Sphere
+        const Scalar m_density;              //!< Solvent density
+        unsigned int m_seed;                 //!< Seed to evaporator pseudo-random number generator
+        const mpcd::detail::boundary m_bc;   //!< Boundary condition
+        unsigned int m_Npick;                              //!< Number of particles picked for evaporation on this rank
+
+        GPUArray<unsigned int> m_bounced_index;           //!< Indices of bounced particles
+        GPUVector<unsigned int> m_picks;                   //!< Particles picked for evaporation on this rank
+
+        virtual void applyPicks();                          //!< Apply the picks
+
+    private:
+        std::vector<unsigned int> m_all_picks;             //!< All picked particles on all the ranks
+
+        //!< For Making a random pick of particles across all ranks
+        void makeAllPicks(unsigned int timestep, unsigned int N_pick, unsigned int N_bounced_total);
+
+    };
+
+/*!
+ * \param timestep Current time to stream
+ */
+
+void DryingDropletStreamingMethod::stream(unsigned int timestep)
+    {
+    /*
+     * Stream will only happen at streaming period and
+     * geometry will only be updated when stream will happen
+     */
+    if(timestep%m_period != 0) return;
+    
+    //compute final Radius and Velocity of surface
+    const Scalar start_R = m_R->getValue(timestep);
+    const Scalar end_R = m_R->getValue(timestep + m_period);
+    const Scalar V = (end_R - start_R)/(m_mpcd_dt);
+
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry 
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (!m_geom)
+        {
+        m_geom = std::make_shared<mpcd::detail::SphereGeometry>(start_R, V , m_bc ); 
+        validate();
+        m_validate_geom = false;
+        }
+
+    /*
+     *Update the geometry radius to the size at the end of the streaming step.
+     * This needs to be done every time.
+     */
+    m_geom = std::make_shared<mpcd::detail::SphereGeometry>(end_R, V, m_bc );
+    
+    //stream according to base class rules
+    ConfinedStreamingMethod<mpcd::detail::SphereGeometry>::stream(timestep);
+
+    //calculating number of particles to evaporate(such that solvent density remain constant)
+    const unsigned int N_remove = round((4.*M_PI/3.)*end_R*end_R*end_R*m_density);
+    const int N_evap = m_mpcd_pdata->getNGlobal() - N_remove;
+
+    // get the compact array of indexes of bounced particles
```

Comment:
You may want to move this code into its own function, for readability. You can have the function return `N_bounced`.